code/modules/pool/pool_controller.dm

```diff
@@ -0,0 +1,415 @@
+
+/obj/machinery/pool/controller/attackby(obj/item/W, mob/user)
+	if(shocked && !(stat & NOPOWER))
+		shock(user,50)
+	if(stat & (BROKEN))
+		return
+	if(istype(W,/obj/item/reagent_containers))
+		if(W.reagents.total_volume) //check if there's reagent
+			user.visible_message("<span class='boldwarning'>[user] is feeding [src] some chemicals from [W].</span>")
+			if(do_after(user, 50, target = src))
+				for(var/datum/reagent/R in W.reagents.reagent_list)
+					if(R.type in GLOB.blacklisted_pool_reagents)
+						to_chat(user, "[src] cannot accept [R.name].")
+						return
+					if(R.reagent_state == SOLID)
+						to_chat(user, "The pool cannot accept reagents in solid form!.")
+						return
+				reagents.clear_reagents()
+				// This also reacts them. No nitroglycerin deathpools, sorry gamers :(
+				W.reagents.trans_to(reagents, max_beaker_transfer)
+				user.visible_message("<span class='notice'>[src] makes a slurping noise.</span>", "<span class='notice'>All of the contents of [W] are quickly suctioned out by the machine!</span")
+				updateUsrDialog()
+				var/list/reagent_names = list()
+				var/list/rejected = list()
+				for(var/datum/reagent/R in reagents.reagent_list)
+					if((R.volume >= min_reagent_amount) && (!respect_reagent_blacklist || R.can_synth))
+						reagent_names += R.name
+					else
+						reagents.remove_reagent(R.type, INFINITY)
+						rejected += R.name
+				if(length(reagent_names))
+					reagent_names = english_list(reagent_names)
+					var/msg = "POOL: [key_name(user)] has changed [src]'s chems to [reagent_names]"
+					log_game(msg)
+					message_admins(msg)
+				if(length(rejected))
+					rejected = english_list(rejected)
+					to_chat(user, "<span class='warning'>[src] rejects the following chemicals as they do not have at least [min_reagent_amount] units of volume: [rejected]</span>")
+				update_color()
+		else
+			to_chat(user, "<span class='notice'>[src] beeps unpleasantly as it rejects the beaker. Why are you trying to feed it an empty beaker?</span>")
+			return
+	else if(panel_open && is_wire_tool(W))
+		wires.interact(user)
+	else
+		return ..()
+
+/obj/machinery/pool/controller/screwdriver_act(mob/living/user, obj/item/W)
+	. = ..()
+	if(.)
+		return TRUE
+	cut_overlays()
+	panel_open = !panel_open
+	to_chat(user, "You [panel_open ? "open" : "close"] the maintenance panel.")
+	W.play_tool_sound(src)
+	if(panel_open)
+		add_overlay("wires")
+	return TRUE
+
+//procs
+/obj/machinery/pool/controller/proc/shock(mob/user, prb)
+	if(stat & (BROKEN|NOPOWER))		// unpowered, no shock
+		return FALSE
+	if(!prob(prb))
+		return FALSE
+	var/datum/effect_system/spark_spread/s = new /datum/effect_system/spark_spread
+	s.set_up(5, 1, src)
+	s.start()
+	if(electrocute_mob(user, get_area(src), src, 0.7))
+		return TRUE
+	else
+		return FALSE
+
+/obj/machinery/pool/controller/proc/process_reagents()
+	if(last_reagent_process > world.time + reagent_tick_interval)
+		return
+	if(length(reagents.reagent_list) > 0)
```

Comment:
Early return, no need for the `>` operator. You can move the `last_reagent_process = world.time` line before this one.